### PR TITLE
feat(ci): front-end linting ci

### LINKED ISSUE #18: Set up front-end CI
## Describe the issue
Set up Continuous Integration (CI) for the frontend

## Resource request
I will need @claraxuxu to decide which coding style we want to use

## Definition of done
- [x] setup a linter (eslint)
- [x] setup a code formatter
- [x] configure circleci or github actions to automatically check the code on some branches
- [x] setup a testing tool

## testing plan
The CI can be run automatically on some branches 

## JEH
1

### PR DESCRIPTION
**Pull request by TheOrionsB**
# Description

Set up Continuous Integration (CI) for the frontend

# Fixes

- fixes #18 

# Changes include

- [x] New feature | Breaking: ESLint is set up with the airbnb and airbnb-typescript configuration
- [x] New feature: Prettier is set up.
- [x] New feature: vitest is set up and its related folders and files are created.

# Breaking changes

The linter test will fail as the codebase is not compliant with it yet.
yarn.lock file was pushed. @vrn-sh/members should run ``yarn install`` to get the correct versions.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added @vrn-sh/members to the reviewers, or specific members of the team
- [x] I have added the needed labels
- [x] I have tested this code
- [x] I have added / updated tests (unit / functionals / end-to-end / ...)
- [x] I have updated the README and other relevant documents (guides...)

# Additional comments

Will need @bogdzn to set up the github workflow that runs the linter and tests on pull requests

### CODE REVIEW

**Comment by TheOrionsB** [comment]
Finally, done. Sorry for all the commits. Couldn't figure out how to test the Github actions without pushing.
Linting fails as it's supposed to considering the codebase is not yet compliant with the linter.

File: .github/workflows/frontend-lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    branches: [ "*" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Comment (djnnvx) [inline .github/workflows/frontend-lint.yml]: Current `node` maintained version is `14.18`. Please see [tool-versions](https://github.com/vrn-sh/erp/blob/18-set-up-front-end-ci/.tool-versions) & [asdf](https://asdf-vm.com/guide/getting-started.html) for more info

File: .github/workflows/frontend-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    branches: [ "*" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Comment (djnnvx) [inline .github/workflows/frontend-unit-tests.yml]: Same here, version 16.x and 18.x are unneeded

File: web/.eslintrc.cjs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (djnnvx) [inline web/.eslintrc.cjs:1]: Please exclude `dist/` folder

**Review by djnnvx — CHANGES_REQUESTED** [review]
Couple things are missing, but overall good PR. Let's have a call tomorrow night for the missing CI.

**Comment by TheOrionsB** [comment]
safe to be merged:
Team.tsx linter errors while be fixed when #43  will be merged

**Review by djnnvx — APPROVED** [review]

**Pull request merged**